src/02-common-principles.md

```diff
@@ -149,6 +149,11 @@ data from subject `01`.
 The suffix `eeg` and the extension `.edf` depend on the imaging modality and
 the data format and indicate further details of the file's contents.
 
+Key-value pairs SHOULD NOT be used to encode scanning parameters of the
+corresponding data.
```

Comment:
Because a large part of these changes are just a refactor of current specification about fieldmaps, I stripped those changes out to #651, so that the actual changes of this new feature were more clear in a diff.

In particular, regarding this particular change:

- This message has been up-ranked to common principles: https://github.com/bids-standard/bids-specification/pull/651/files#diff-78a960934ce25feed2c29612b842d25131d0360556c2afc4dcaa9b60aceeb2baR152-R156
- A note has been at this point, linking to the common principles: https://github.com/bids-standard/bids-specification/pull/651/files#diff-e8c70f785c264706f1f453d744d53e74d7d4b77a5bbe25dc836adf6b96b36d50R719-R721

I believe it would be best following up this conversation in #651 (I'll rebase this from time to time).

Reply:
This is not to say I disagree with your suggestion, please let me know any changes you would suggest along what you pointed out, but in the context of the refactor PR.